index.js

```diff
@@ -325,24 +326,63 @@ if(!helpers.isTestEnvironment()){
 }
 
 // Handler for income SmartThings POST requests
-app.post('/api/st', function(req, res, next) {    // eslint-disable-line no-unused-vars -- next might be used in the future
-    smartapp.handleHttpCallback(req, res);
-});
+app.post('/api/st', Validator.body(['lifecycle']).exists(), function(req, res, next) {    // eslint-disable-line no-unused-vars -- next might be used in the future
+    try {
+        const validationErrors = Validator.validationResult(req)
+  
+        if(validationErrors.isEmpty()){
+            smartapp.handleHttpCallback(req, res)
+        } else {
+            helpers.log(`Bad request, parameters missing ${JSON.stringify(validationErrors)}`)
+            res.status(400).send()
+        }
+    }
+    catch(err) {
+        helpers.log(err)
+        res.status(500).send()
+    }
+})
 
 // Handler for income XeThru POST requests
-app.post('/api/xethru', async (req, res) => {
-    // eslint-disable-next-line no-unused-vars -- might be useful in the future to know what all we have access to in the body
-    const {deviceid, locationid, devicetype, state, rpm, distance, mov_f, mov_s} = req.body;
-
-    redis.addXeThruSensorData(req, res);
-    handleSensorRequest(locationid);
-});
+app.post('/api/xethru', Validator.body(['locationid']).exists(), async (req, res) => {
```

Comment:
This also needs `state`, `rpm`, `distance`, `mov_f`, and `mov_s` in the `body` as used in `redis.addXeThruSensorData`.

Reply:
Actually, now that I'm looking at this... if you're up for even more refactoring, it would be really nice if these route functions were the only ones to know about the request and response objects. To do this, functions that are called from them, like `redis.addXeThruSensorData(req, res)` would need to change to take in `locationid, state, rpm, distance, mov_f, mov_s` as separate parameters and not send back the `200` response.

This would be good for a bunch of reasons...
1. It would make it easier to see exactly what is being used and why it's being validated here.
2. It would make the called functions easier to test because you wouldn't need to match the structure of the request object or mock a response object
3. Would be generally good separation of concerns (i.e. why should the Redis module need to care that the data it was getting came from an API endpoint)

However, this change isn't strictly related to your Task. So it's up to you. It would be great if you could do the easy ones, but if it starts being tricky (like for how `smartapp.handleHttpCallback` sends the request off to `signature.signatureIsVerified` for even more processing), then don't bother with those ones.

Reply:
Definitely something I can look into, but yeah as you point out I'm not sure I want to mess around with functions from other libraries where stuff is obfuscated like in `handleHttpCallback`

Reply:
Also this lines up with the linting because it doesn't like that random object that is just `= req.body` and then most of what's inside it goes unused

Reply:
@trisapeace can you please confirm that what we'd be wanting in this case would be something like

```
if(validationErrors.isEmpty()){
    const { locationid, state, rpm, distance, mov_f, mov_s } = req.body

    redis.addXeThruSensorData(locationid, state, rpm, distance, mov_f, mov_s)
    handleSensorRequest(locationid)

    res.status(200).send()
} 
```